pkg/pod/status_test.go

```diff
@@ -1276,6 +1434,11 @@ func TestMakeTaskRunStatus(t *testing.T) {
 			// Common traits, set for test case brevity.
 			c.want.PodName = "pod"
 			c.want.StartTime = &metav1.Time{Time: startTime}
+			for i := range c.want.TaskRunStatusFields.Steps {
+				if c.want.TaskRunStatusFields.Steps[i].Results == nil {
+					c.want.TaskRunStatusFields.Steps[i].Results = []v1.TaskRunResult{}
+				}
+			}
```

Comment:
I'm guessing this is needed to handle `nil` vs `{}`?

Not super important, but how annoying would it be to properly return nil values for cases where there are no results?

Reply:
Yes, that's for nil vs {}.

Let me look into returning nil values.